code/game/machinery/bots/farmbot.dm

```diff
@@ -1,597 +0,0 @@
-//Farmbots by GauHelldragon - 12/30/2012
-// A new type of buildable aiBot that helps out in hydroponics
-
-// Made by using a robot arm on a water tank and then adding:
-// A plant analyzer, a bucket, a mini-hoe and then a proximity sensor (in that order)
-
-// Will water, weed and fertilize plants that need it
-// When emagged, it will "water", "weed" and "fertilize" humans instead
-// Holds up to 10 fertilizers (only the type dispensed by the machines, not chemistry bottles)
-// It will fill up it's water tank at a sink when low.
-
-// The behavior panel can be unlocked with hydroponics access and be modified to disable certain behaviors
-// By default, it will ignore weeds and mushrooms, but can be set to tend to these types of plants as well.
-
-//Seems a little stupid handling multiple Go direction X,Y,Z to do Job F,G,H if it's in multiple directions so I'd reccomend only enabling 1 water/fertilize/weed task at a time. - SarahJohnson
-
-
-#define FARMBOT_MODE_WATER			1
-#define FARMBOT_MODE_FERTILIZE	 	2
-#define FARMBOT_MODE_WEED			3
-#define FARMBOT_MODE_REFILL			4
-#define FARMBOT_MODE_WAITING		5
-
-#define FARMBOT_ANIMATION_TIME		25 //How long it takes to use one of the action animations
-#define FARMBOT_EMAG_DELAY			60 //How long of a delay after doing one of the emagged attack actions
-#define FARMBOT_ACTION_DELAY		35 //How long of a delay after doing one of the normal actions
-
-/obj/machinery/bot/farmbot
-	name = "Farmbot"
-	desc = "The botanist's best friend."
-	icon = 'icons/obj/aibots.dmi'
-	icon_state = "farmbot0"
-	layer = 5.0
-	density = 1
-	anchored = 0
-	health = 50
-	maxhealth = 50
-	req_access =list(access_hydroponics)
-
-	var/Max_Fertilizers = 10
-
-	var/setting_water = 1
-	var/setting_refill = 1
-	var/setting_fertilize = 1
-	var/setting_weed = 1
-	var/setting_ignoreWeeds = 1
-	var/setting_ignoreMushrooms = 1
-
-	var/atom/target //Current target, can be a human, a hydroponics tray, or a sink
-	var/mode //Which mode is being used, 0 means it is looking for work
-
-	var/obj/structure/reagent_dispensers/watertank/tank // the water tank that was used to make it, remains inside the bot.
-
-	var/path[] = new() // used for pathing
-	var/frustration
```

Comment:
farmbots gonna get all frustrated and fertilize the SHIT out of you....then collect it...to fertilize so more...